src/libasr/pass/replace_symbolic.cpp

```diff
@@ -0,0 +1,175 @@
+                        nullptr, nullptr, ASR::storage_typeType::Default, arg_type, nullptr,
+                        ASR::abiType::BindC, ASR::Public, ASR::presenceType::Required, true));
+                    fn_symtab->add_symbol(s2c(al, "x"), arg);
+                    args.push_back(al, ASRUtils::EXPR(ASR::make_Var_t(al, x.base.base.loc, arg)));
+                }
+                Vec<ASR::stmt_t*> body;
+                body.reserve(al, 1);
+
+                Vec<char *> dep;
+                dep.reserve(al, 1);
+
+                ASR::asr_t* new_subrout = ASRUtils::make_Function_t_util(al, x.base.base.loc,
+                    fn_symtab, s2c(al, new_name), dep.p, dep.n, args.p, args.n, body.p, body.n,
+                    nullptr, ASR::abiType::BindC, ASR::accessType::Public,
+                    ASR::deftypeType::Interface, nullptr, false, false, false,
+                    false, false, nullptr, 0, nullptr, 0, false, false, false, s2c(al, header));
+                ASR::symbol_t *new_symbol = ASR::down_cast<ASR::symbol_t>(new_subrout);
+                current_scope->add_symbol(new_name, new_symbol);
+                symbolic_replaces_with_CPtr_Module = false;
+            }
+        }
+        current_scope = current_scope_copy;
+    }
+
+    void visit_Function(const ASR::Function_t& x) {
+        ASR::Function_t &xx = const_cast<ASR::Function_t&>(x);
+        SymbolTable* current_scope_copy = current_scope;
+        current_scope = xx.m_symtab;
+        for (auto &item : current_scope->get_scope()) {
+            if (is_a<ASR::Variable_t>(*item.second)) {
+                this->visit_symbol(*item.second);
+                if(symbolic_replaces_with_CPtr_Function){
+                    std::string var = std::string(item.first);
+                    std::string placeholder = "_" + var;
+                    ASR::symbol_t* var_sym = current_scope->get_symbol(var);
+                    ASR::symbol_t* placeholder_sym = current_scope->get_symbol(placeholder);
+                    ASR::expr_t* target1 = ASRUtils::EXPR(ASR::make_Var_t(al, xx.base.base.loc, placeholder_sym));
+                    ASR::expr_t* target2 = ASRUtils::EXPR(ASR::make_Var_t(al, xx.base.base.loc, var_sym));
+
+                    // statement 1
+                    int cast_kind = ASR::cast_kindType::IntegerToInteger;
+                    ASR::ttype_t *type1 = ASRUtils::TYPE(ASR::make_Integer_t(al, xx.base.base.loc, 4));
+                    ASR::ttype_t *type2 = ASRUtils::TYPE(ASR::make_Integer_t(al, xx.base.base.loc, 8));
+                    ASR::expr_t* cast_tar = ASRUtils::EXPR(ASR::make_IntegerConstant_t(al, xx.base.base.loc, 0, type1));
+                    ASR::expr_t* cast_val = ASRUtils::EXPR(ASR::make_IntegerConstant_t(al, xx.base.base.loc, 0, type2));
+                    ASR::expr_t* value1 = ASRUtils::EXPR(ASR::make_Cast_t(al, xx.base.base.loc, cast_tar, (ASR::cast_kindType)cast_kind, type2, cast_val));
```

Comment:
This can be initialized in the sym table declaration itself.
Just pass `ASRUtils::EXPR(ASR::make_IntegerConstant_t(al, xx.base.base.loc, 0, type1))` to `m_symbolic_value` and `m_value`.